### PR TITLE
Court outcome letter template

### PR DESCRIPTION
**Pull request by Miles-Alford**
## Context

When a court outcome is created a letter needs to be sent to the tenant informing them of that outcome

Generic outcome
Screenshot 2020-09-14 at 15 45 22

Outcome with terms
Screenshot 2020-09-14 at 15 46 51

Outcome with outright possession
Screenshot 2020-09-14 at 15 49 20

## Changes in this pull request

- added court outcome letter template 
- added the ability to preview the court outcome letter 

## Things to check
- [ ] This code doesn't rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] Environment variables have been updated

### CODE REVIEW

File: app/models/hackney/income_collection/letter/court_outcome.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        TEMPLATE_PATHS = [
+          'lib/hackney/pdf/templates/income/court_outcome_letter.erb'
+        ].freeze
+        MANDATORY_FIELDS = %i[court_outcome].freeze
+
+        attr_reader :court_outcome, :court_hearing_arrears, :instalment_amount, :frequency, :rent, :weekly_payment, :date_of_first_payment
+
+        def initialize(params)
+          super(params)
+
```
Comment (elena-vi) [inline app/models/hackney/income_collection/letter/court_outcome.rb:16]:
```ruby
mandatory_for_this_letter = MANDATORY_FIELDS + [EXTRA_MANDATORY_FIELDS] if formal_agreement?
validated_params = validate_mandatory_fields(mandatory_for_this_letter, params)
```

File: lib/hackney/pdf/income_preview.rb [hunk 2]
```diff
@@ -7,23 +7,15 @@ def initialize(get_templates_gateway:, income_information_gateway:, tenancy_case
...
 
-        if agreement
+        letter_params = income_info
 
-          if agreement.breached?
-            agreement_data = agreement.formal? ? get_breached_formal_agreement_info(agreement) : get_breached_agreement_info(agreement)
-          else
-            agreement_data = get_agreement_info(agreement)
-          end
+        letter_params = letter_params.merge(court_outcome_params(agreement, court_case, income_info)) if court_case
```
Comment (Ninamma) [inline lib/hackney/pdf/income_preview.rb:16]: 😍 I likeeee

File: lib/use_cases/generate_and_store_letter.rb [hunk 3]
```diff
@@ -9,6 +9,8 @@ def execute(payment_ref:, tenancy_ref:, template_id:, user:)
       income_collection_templates = %w[income_collection_letter_1 income_collection_letter_2]
       agreement_templates = %w[informal_agreement_confirmation_letter informal_agreement_breach_letter formal_agreement_breach_letter]
 
+      court_case_templates = %(court_outcome_letter)
```
Comment (Ninamma) [inline lib/use_cases/generate_and_store_letter.rb:12]: Is `%(court_outcome_letter)` the same as `%w[court_outcome_letter]`? 🤔

Comment (elena-vi) [reply lib/use_cases/generate_and_store_letter.rb:12]:
```suggestion
      court_case_templates = %[court_outcome_letter]
```

Comment (elena-vi) [reply lib/use_cases/generate_and_store_letter.rb]: not sure, but I want to change that for consistency

**Review by Ninamma — APPROVED** [review]
Looks good to me! And will be helpful for the Court Date letter too, thanks! 

One thing I was confused by when working on the Court Date letter was what the purpose of the tests in `spec/lib/hackney/pdf/income_preview_spec.rb` was 🤔 I don't think I understand what they do - maybe can chat about it tomorrow?

**Pull request merged**